### DISCUSSION TITLE
Clean install -- PowerShell no longer working

### DISCUSSION THREAD

**Discussion by juancrl** [dotnet/interactive#2068] [Q&A]
Apologies for the ambiguities, I'm no expert in notebooks.   (x-posted in vscode-Jupyter)

I had several *.ipynb files with a mix of PowerShell code and markdown. Working perfectly until last week. Visual Studio Code insiders, .NET interactive notebooks, Jupyter notebooks, several kernels, all updated to the latest version.
Then there was an update on Visual Studio Code Insiders version, which probably triggered some updates on these extensions.
Now all notebooks give errors. I can only select a Python Kernel... even if the cells are marked as PowerShell, they seem to be interpreted as Python code. No PowerShell kernel at all. pwsh 7.2.4 running ok outside the notebooks.

I've decided to start fresh, uninstalled all extensions, VSCode, deleted all evident related files, rebooted, and installed all again. VSCode insiders, then Python, then PowerShell extension, then .NET Interactive which installs this Jupyter extension.
But after this, same situation, and only Python kernel available in the picker.

Another symptom : ".NET Interactive : Create new blank notebook" display "Activate Extensions"... and does nothing.

What is the right procedure (documentation of) to install the simplest environment which can run PowerShell, markdown, and maybe cells with C# and Python ?

**Comment by jonsequitur** [comment]
The extension alone should work without needing to install anything  else beyond the .NET 6 SDK. I would recommend using stable VS Code rather than Insiders unless there's a specific new feature you're trying out.

**Reply by juancrl** [reply]
Thanks. I uninstalled both VSCODE editions (this time using Appwiz.CPL... and as they're not there, looking for Uninst_001.exe or similar in their folders. This did remove them, not just cleaning their folders)
Then installed only VSCODE and .NET interactive from scratch. Seems to be operative now, at least with PowerShell+Markdown.
Worried... as this was working perfectly before the last update.

**Reply by jonsequitur** [reply]
You shouldn't need to uninstall VS Code. But keep your eye out for update notifications, either for VS Code (if you see a badge on the gear icon in the lower left) or a "Reload required" badge on the extension itself.

**Reply by juancrl** [reply]
Sure. But when you feel something is corrupt, two sXs installations of VSCode, the .NET Kernel appears randomly as "Python" -with the Python executable path-, and an update removes all kernel definitions (from the undocumented place) and all your scripts used in all your work demos fail suddenly...  After all updates and reloads... Troubleshooting is useless. Clean install is faster (practical but disappointing)
BTW, I'm running SysInternals Process Monitor to find out something simple... where the list of defined Kernels is stored. No luck.

**Reply by jonsequitur** [reply]
.NET Interactive can act as a Jupyter kernel or it can be used directly without Jupyter. In the latter case, you don't need anything  installed other than the .NET Interactive Notebooks VS Code extension and the .NET SDK.

The Jupyter extension (which the .NET Interactive Notebooks  extension depends on) will pick up Jupyter kernels that are installed on your machine. 

<img width="607" alt="image" src="https://user-images.githubusercontent.com/547415/173856739-4199d5ff-8d32-4cfc-9764-cecd920cbfc6.png">

These will also show up in kernel quick pick that you see when you click the kernel button  the upper right corner of the notebook.

<img width="704" alt="image" src="https://user-images.githubusercontent.com/547415/173856929-83b8901a-1729-404e-82ff-7a60866c19b4.png">

.NET Interactive can be registered  as a Jupyter kernel, which is why you see listings in the  quick pick under both "Jupyter kernel" and "Other".

The "Other" option here chooses .NET Interactive via its own protocol rather than Jupyter,  so  you don't need machine-level installation of Jupyter and registration of kernels.

**Reply by juancrl** [reply]
Thanks ! Now thinks are clearer. 
To learn more, which kit/app provides that Jupyter.exe ? I don't have that on my environment, or Get-Command doesn't find it. I assume it has options to list but also add Kernels.
